spark/interpreter/src/main/java/org/apache/zeppelin/spark/KotlinSparkInterpreter.java

```diff
@@ -0,0 +1,198 @@
+          KOTLIN_SPARK_SUPPORTED_VERSION + " and newer are supported");
+    }
+
+    z = sparkInterpreter.getZeppelinContext();
+
+    SparkKotlinReceiver ctx = new SparkKotlinReceiver(
+        sparkInterpreter.getSparkSession(),
+        jsc,
+        sparkInterpreter.getSQLContext(),
+        z);
+
+    List<String> classpath = sparkClasspath();
+
+    String outputDir = null;
+    SparkConf conf = jsc.getConf();
+    if (conf != null) {
+      outputDir =  conf.getOption("spark.repl.class.outputDir").getOrElse(null);
+    }
+
+    interpreter.getKotlinReplProperties()
+        .receiver(ctx)
+        .classPath(classpath)
+        .outputDir(outputDir)
+        .codeOnLoad(KotlinZeppelinBindings.Z_SELECT_KOTLIN_SYNTAX)
+        .codeOnLoad(KotlinZeppelinBindings.SPARK_UDF_IMPORTS)
+        .codeOnLoad(KotlinZeppelinBindings.CAST_SPARK_SESSION);
+    interpreter.open();
+  }
+
+  @Override
+  public void close() throws InterpreterException {
+    interpreter.close();
+  }
+
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context)
+      throws InterpreterException {
+
+    if (isUnsupported()) {
+      return unsupportedMessage;
+    }
+
+    z.setInterpreterContext(context);
+    z.setGui(context.getGui());
+    z.setNoteGui(context.getNoteGui());
+    InterpreterContext.set(context);
+
+    jsc.setJobGroup(buildJobGroupId(context), buildJobDesc(context), false);
+    jsc.setLocalProperty("spark.scheduler.pool", context.getLocalProperties().get("pool"));
+
+    InterpreterOutput out = context.out;
+    PrintStream scalaOut = Console.out();
+    PrintStream newOut = (out != null) ? new PrintStream(out) : null;
+
+    Console.setOut(newOut);
+    InterpreterResult result = interpreter.interpret(st, context);
+    Console.setOut(scalaOut);
+
+    return result;
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) throws InterpreterException {
+    if (isUnsupported()) {
+      return;
+    }
+    jsc.cancelJobGroup(buildJobGroupId(context));
+    interpreter.cancel(context);
+  }
+
+  @Override
+  public FormType getFormType() throws InterpreterException {
+    return interpreter.getFormType();
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) throws InterpreterException {
+    if (isUnsupported()) {
+      return 0;
+    }
+    return sparkInterpreter.getProgress(context);
+  }
+
+  @Override
+  public List<InterpreterCompletion> completion(String buf, int cursor,
+      InterpreterContext interpreterContext) throws InterpreterException {
+    if (isUnsupported()) {
+      return Collections.emptyList();
+    }
+    return interpreter.completion(buf, cursor, interpreterContext);
+  }
+
+  private boolean isUnsupported() {
+    return unsupportedMessage != null;
+  }
+
+  private List<String> sparkClasspath() {
+    String sparkJars = System.getProperty("spark.jars");
+    Pattern isKotlinJar = Pattern.compile("/kotlin-(runtime|stdlib|compiler|reflect)(-.*)?\\.jar");
```

Comment:
May be this pattern simplified to `/kotlin-[a-z]*(-.*)?\\.jar`? I suppose we don't want to change this regex every time new jar is added.

Reply:
fixed